### ISSUE TITLE
Class transformer not working with swc

### ISSUE THREAD

**Issue by DanielRamosAcosta** [swc-project/swc#2117]
**Describe the bug**

When applying the `plainToClass` function to a class defined with `class-validator`, it returns the class with no attributes

**Input code**

[Here is a repo with a test reproducing the error](https://github.com/DanielRamosAcosta/swc-class-transformer-bug)

```ts
const usersOptionsInput = plainToClass(GetUsersOptionsInput, {
  query: "Query",
  pagination: { skip: 0, limit: 20 },
});

expect(usersOptionsInput.query).toEqual("Query"); // error
```

**Config**

You can also find the config at the [repo reproducing the error](https://github.com/DanielRamosAcosta/swc-class-transformer-bug/blob/main/.swcrc) 

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false,
      "decorators": true
    },
    "transform": {
      "legacyDecorator": true,
      "decoratorMetadata": true
    },
    "target": "es2018"
  },
  "module": {
    "type": "commonjs",
    "noInterop": true
  }
}
```

**Expected behavior**
I would expect the instance of the class to have all the defined attributes.

**Version**
The version of @swc/core: 1.2.80

**Additional context**

Just one more time, [I created this repo in order to reproduce the error](https://github.com/DanielRamosAcosta/swc-class-transformer-bug)

This error is related to #1362

**Comment by unlight** [comment]
Issue on `class-transformer` side https://github.com/typestack/class-transformer/issues/796 I suppose invalid, and it should be fixed in swc.

**Comment by tonivj5** [comment]
@unlight if class-validator works perfect with tsc and fail with swc. I think it's a problem of swc handling decorators...

**Comment by kdy1** [comment]
@tonivj5 Did you try importing `reflect-metadata`?

**Comment by tonivj5** [comment]
@kdy1 yep! It doesn't work.

This code does not work:
```ts
import "reflect-metadata";
import { IsOptional, IsString } from "class-validator";

export class GetUsersOptionsInput {
  @IsString()
  @IsOptional()
  public query?: string;
}
```

This code does work:
```ts
import "reflect-metadata";
import { IsOptional, IsString } from "class-validator";

export class GetUsersOptionsInput {
  @IsString()
  @IsOptional()
  public query?: string = undefined;
}
```

I did a deeper research here https://github.com/swc-project/swc/issues/2117#issuecomment-906681274

**Comment by kshahani** [comment]
@tonivj5 Thank you so much for that workaround!

@kdy1 Fixing this would make a world of difference for anyone part of the Nest.js community, it'd be hugely appreciated!

**Comment by charlescapps** [comment]
+1, it would be great to fix this. We use `class-validator` pretty heavily, and we would like to run our unit tests with `ts-node + swc` since it's dramatically faster than using the default transpiler with `ts-node`.

**Comment by RIP21** [comment]
If you're struggling with this issue here is the way to do it so it's almost 0 pain while we're waiting for fix.
https://github.com/nestjs/nest-cli/issues/731#issuecomment-993936178

**Comment by acoroleu-tempus** [comment]
this is blocking for us to migrate to swc, company wide

**Comment by RIP21** [comment]
@acoroleu-tempus use the workaround I mentioned above.

**Comment by kdy1** [comment]
I investigated this.

For the `initializer` property of the property descriptor for fields without an initializer, babel uses `null` while swc emits `void 0`. `_applyDecoratedDescriptor` returns `null` if `initializer` is `undefined`, so the property descriptor used for `_initializerDefineProperty` while instantiating a class is an object for babel while being `null` for swc.

https://github.com/babel/babel/blob/69ae3b5a250d60618702751aa2aa4f56ccd43988/packages/babel-plugin-proposal-decorators/src/transformer-legacy.ts#L146-L152

I tried changing the value to `null`, but it broke other tests. Namely, the tests I added for https://github.com/swc-project/swc/issues/2127 in https://github.com/swc-project/swc/pull/3105 was problematic. 

I decided to postpone this.
